assets/js/upload.js

```diff
@@ -47,32 +53,150 @@ function setupImageUpload() {
       label.appendChild(imgWrap);
       imgPreviews.appendChild(label);
     });
+
+    validationChecks.image = true;
+    updateValidation();
   }
   function showError() {
     clearEl(imgPreviews);
     showEl(scraperError);
     enableFetch();
+
+    validationChecks.image = false;
+    updateValidation();
   }
   function hideError()    { hideEl(scraperError); }
+
+  function showImageError(text) {
```

Comment:
As noted above, I'd probably split the functionality of _showing validation errors_ separately from the rest of the upload logic.